### DISCUSSION TITLE
Include user-friendly versions of geolocation data in API results

### DISCUSSION THREAD

**Discussion by Devvypaws** [plausible/analytics#1810] [Anything Else]
## Feature Request

This request is composed of two related features:
1. Provide country, region, and city names in user-friendly formats when their respective data is included in API responses
2. Include the country code in API responses where region or city data is returned

An alternative could be an endpoint that can provide this information when given a geolocation code. This data would be cacheable on the requesting site's side to reduce load on the API server if that is a concern; we're not knowledgeable on the performance characteristics of Clickhouse.

## Background

We're developing a plugin for a CMS that currently uses the official cloud-hosted API to provide data for an analytics dashboard. While installing a mapping of country codes in the database would be reasonable, — maybe even regions — the entirety of the cities data set is too large and an unreasonable ask for site owners. We cannot know a site's capabilities upfront, both processing- and storage-wise, and would like to make this plugin usable by as many site owners as possible to encourage the switch from other analytics platforms.

In regard to part 2 of the request, our motivation for this is to be able to provide country flags alongside regions and cities, similar to the official Plausible dashboard.

**Comment by ukutaht** [comment]
Thanks for the request!

Good point for sure, the Geonames database we use to fetch display names for cities takes about 1GB of memory. The data you requested can be added fairly easily. No worries about Clickhouse load or anything like that. All the data is already in memory.
